features/opennms-es-rest/src/main/java/org/opennms/plugins/elasticsearch/rest/EventForwarderQueueImpl.java

```diff
@@ -117,9 +122,13 @@ public EventIndexProducer(int batchSize, int batchInterval) {
 		 */
 		@Override
 		public void dispatch(List<Event> events) {
-			// Ensure we are initialized correctly.
-			if (!elasticSearchInitializer.isInitialized()) {
-				elasticSearchInitializer.initialize(); // blocks until initialized properly
+			try {
+				// Ensure we are initialized correctly.
+				if (!elasticSearchInitializer.isInitialized()) {
+					elasticSearchInitializer.initialize(); // blocks until initialized properly
+				}
+			} catch (ConnectionPoolShutdownException ex) { // Connection Pool is gone, nothing we can do
+				ExceptionUtils.handle(getClass(), ex, events);
```

Comment:
Should return afterwards. There is no need in forwarding the events afterwards, as that will fail as well.